src/components/multiSelect/MultiSelectBottomSheet.tsx

```diff
@@ -0,0 +1,271 @@
+      Keyboard.dismiss()
+    }
+
+    if (toIndex === 0) {
+      onOpen?.()
+    }
+  }
+  const handleClose = () => {
+    onClose?.()
+  }
+
+  // Multi select logic things
+  const isEveryItemSelected = Object.values(selectedItems).every((isSelected) => isSelected)
+  const allItemIds = Object.keys(selectedItems) as T[]
+  const everyItemSelected = useMemo(
+    () =>
+      allItemIds.reduce(
+        (acc, key) => {
+          acc[key as T] = true
+          return acc
+        },
+        {} as Record<T, boolean>
+      ),
+    [allItemIds]
+  )
+  const noItemSelected = useMemo(
+    () =>
+      allItemIds.reduce(
+        (acc, key) => {
+          acc[key as T] = false
+          return acc
+        },
+        {} as Record<T, boolean>
+      ),
+    [allItemIds]
+  )
+
+  const selectAllItem = renderItem({
+    text: selectAllText,
+    isSelected: isEveryItemSelected,
+    onPress: () => setSelectedItems(everyItemSelected),
+  })
+
+  const items = (Object.entries(selectedItems) as [T, boolean][]).map(([itemId, isSelected]) =>
+    renderItem({
+      text: textAndIconMap[itemId].text,
+      iconUrl: textAndIconMap[itemId].iconUrl,
+      isSelected: isSelected && !isEveryItemSelected,
+      onPress: () => {
+        if (isEveryItemSelected) {
+          setSelectedItems({ ...noItemSelected, [itemId]: isSelected })
+        } else {
+          setSelectedItems({ ...selectedItems, [itemId]: !isSelected })
+        }
+      },
+    })
+  )
+
+  return (
+    <GorhomBottomSheet
+      ref={forwardedRef}
+      index={-1}
+      enableDynamicSizing={true}
+      enablePanDownToClose
+      backdropComponent={renderBackdrop}
+      handleIndicatorStyle={{ width: 0 }}
+      onAnimate={handleAnimate}
+      onClose={handleClose}
+      maxDynamicContentSize={height - insets.top}
+      backgroundStyle={{ backgroundColor: 'transparent' }}
+    >
+      <BottomSheetScrollView
+        forwardedRef={scrollViewRef}
+        testId={'MultiSelectBottomSheet'}
+        containerStyle={styles.bottomSheetScrollView}
+      >
+        <View style={[styles.item, styles.borderRadiusTop]}>
+          <Text style={styles.boldTextStyle}>{title}</Text>
+        </View>
+        <View style={styles.itemsContainer}>
+          <ScrollView style={{ height: ITEM_HEIGHT * MAX_ITEMS_IN_VIEW }}>
+            {selectAllItem}
+            {items}
+          </ScrollView>
+        </View>
+        <View style={styles.doneButtonContainer}>
+          <Touchable
+            testID="MultiSelectBottomSheet/Done"
+            style={styles.doneButton}
+            onPress={handleClose}
+          >
+            <Text style={styles.boldTextStyle}>{t('done')}</Text>
+          </Touchable>
+        </View>
+      </BottomSheetScrollView>
+    </GorhomBottomSheet>
+  )
+}
+
+export interface ItemProps {
+  onPress?: () => void
+  text: string
+  isSelected: boolean
+  iconUrl?: string
+}
+function renderItem({ onPress, text, iconUrl, isSelected }: ItemProps) {
```

Comment:
nit: could you make this a react component. E.g., function Item(...) { }, and then do `<Item ...>` above, think it will make the above a bit more readable.